numpy/core/numeric.py

```diff
@@ -1451,6 +1451,10 @@ def moveaxis(a, source, destination):
     (5, 4, 3)
 
     """
+    # Fast-path for the no-op case when source and destination are small
+    # (cached) integers.  Relevant speedup e.g. for quantile.
+    if source is destination and isinstance(a, ndarray):
+        return a
```

Comment:
In retrospect this probably isn't a good idea after all, as it make `np.moveaxis(a, 100, 100)` legal, and changes the value of `np.moveaxis(a, 0, 0) is a`.

Reply:
So you want to move back to special casing axis=0 in quantile?  (I'm fine either way.)

Reply:
Yeah, I guess that's probably best after all, assuming it actually makes a measurable difference

Reply:
back to the old version.